docs/se/oci/02_object-storage.adoc

```diff
@@ -0,0 +1,203 @@
+== Using the Object Storage
+
+In the Service we must specify the mapping for CRUD operations with the files and their handlers:
+
+[source,java]
+----
+@Override
+public void update(Routing.Rules rules) {
+    rules.get("/file/{file-name}", this::download)
+            .post("/file/{file-name}", this::upload)
+            .delete("/file/{file-name}", this::delete)
+            .get("/rename/{old-name}/{new-name}", this::rename);
+}
+----
+
+To upload a file via `PUT` method to OCI Object Storage:
+
+[source,java]
+----
+private void upload(ServerRequest req, ServerResponse res) {
+    OptionalLong contentLength = req.headers().contentLength();
+    if (contentLength.isEmpty()) {
+        req.content().forEach(DataChunk::release);
+        res.status(Http.Status.BAD_REQUEST_400).send("Content length must be defined");
+        return;
+    }
+
+    String objectName = req.path().param("file-name");
+
+    PutObject.Request request = PutObject.Request.builder() <1>
+            .objectName(objectName)
+            .bucket(bucketName)
+            .contentLength(contentLength.getAsLong());
+
+    req.headers().contentType().ifPresent(request::requestMediaType); <2>
+
+    objectStorage.putObject(request,
+                            req.content())
+            .forSingle(response -> res.send(response.requestId())) <3>
+            .exceptionally(res::send);
+}
+----
+
+<1> Create the `Request` using `PutObject.Request.builder()`
+<2> Define `MediaType`
+<3> Execute the request to OCI in asynchronous way and put the result in `response` object
+
+To download the file via `GET` method from OCI Object Storage:
+
+[source,java]
+----
+private void download(ServerRequest req, ServerResponse res) {
+    String objectName = req.path().param("file-name");
+
+    objectStorage.getObject(GetObject.Request.builder()
+                                    .bucket(bucketName)
+                                    .objectName(objectName)) <1>
+            .forSingle(apiResponse -> {
+                Optional<GetObjectRx.Response> entity = apiResponse.entity(); <2>
+                if (entity.isEmpty()) {
+                    res.status(Http.Status.NOT_FOUND_404).send(); <3>
+                } else {
+                    GetObjectRx.Response response = entity.get();
+                    // copy the content length header to response
+                    apiResponse.headers()
+                            .first(Http.Header.CONTENT_LENGTH)
+                            .ifPresent(res.headers()::add);
+                    res.send(response.publisher()); <4>
+                }
+            })
+            .exceptionally(res::send);
+}
+----
+
+<1> Use `getObject` function to make asynchronous request to OCI Object Storage
+<2> The result is of type `Optional`
+<3> Whenever the result is empty, return status `404`
+<4> Get the response, set headers and return the result as a `Publisher`
+
+
+To rename an existing file in the OCI bucket a `GET` method with two parameters:
```

Comment:
To rename an existing file in the OCI bucket, submit a `GET` method with two parameters:

Reply:
fixed